lib/Github/HttpClient/HttpClient.php

```diff
@@ -80,12 +99,25 @@ public function clearHeaders()
 
     public function addListener($eventName, $listener)
     {
-        $this->client->getEventDispatcher()->addListener($eventName, $listener);
+        // Guzzle 3 BC
+        if (method_exists($this->client, 'getEmitter')) {
+            $this->client->getEmitter()->on($eventName, $listener);
+        } else {
+            $this->client->getEventDispatcher()->addListener($eventName, $listener);
+        }
     }
 
-    public function addSubscriber(EventSubscriberInterface $subscriber)
+    /**
+     * @param EventSubscriberInterface|SubscriberInterface $subscriber
+     */
+    public function addSubscriber($subscriber)
     {
-        $this->client->addSubscriber($subscriber);
+        // Guzzle 3 BC
+        if (method_exists($this->client, 'getEmitter')) {
+            $this->client->getEmitter()->attach($subscriber);
```

Comment:
I'm not sure about this but I think it's BC break and this could not be handled.

`attach` method wants a `SubscriberInterface` from Guzzle but `addSubscriber` was aim to get `EventSubscriberInterface` from Symfony.

If this could not be solve, I propose to drop Guzzle 3 support now, start a 1.x branch and make master for 2.0, next major.

@stof What do you think?

I need an answer before continuing this PR.